phoenix-core/src/main/java/org/apache/phoenix/compile/RVCOffsetCompiler.java

```diff
@@ -240,34 +246,41 @@ public CompiledOffset getRVCOffset(StatementContext context, FilterableStatement
             }
         }
 
+        byte[] key;
+
         // check to see if this was a single key expression
         ScanRanges scanRanges = context.getScanRanges();
 
+        //We do not generate a point lookup today in phoenix if the rowkey has a trailing null, we generate a range scan.
         if (!scanRanges.isPointLookup()) {
```

Comment:
I guess pushing the logic for trailing null inside the isPointLookup() API is a bigger change that's too risky?